src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -0,0 +1,190 @@
+package edu.cornell.kfs.fp.batch.service.impl;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.gl.GeneralLedgerConstants;
+import org.kuali.kfs.krad.bo.AdHocRouteRecipient;
+import org.kuali.kfs.krad.document.Document;
+import org.kuali.kfs.krad.exception.ValidationException;
+import org.kuali.kfs.krad.service.DocumentService;
+import org.kuali.kfs.krad.util.ErrorMessage;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.batch.BatchInputFileType;
+import org.kuali.kfs.sys.batch.service.BatchInputFileService;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.sys.document.AccountingDocument;
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.rice.kew.api.exception.WorkflowException;
+import org.springframework.util.AutoPopulatingList;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.batch.service.AccountingDocumentGenerator;
+import edu.cornell.kfs.fp.batch.service.CreateAccountingDocumentService;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentEntry;
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentListWrapper;
+
+public class CreateAccountingDocumentServiceImpl implements CreateAccountingDocumentService {
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CreateAccountingDocumentServiceImpl.class);
+
+    private BatchInputFileService batchInputFileService;
+    private BatchInputFileType accountingDocumentBatchInputFileType;
+    private DocumentService documentService;
+    private ConfigurationService configurationService;
+
+    @Override
+    public void createAccountingDocumentsFromXml() {
+        List<String> inputFileNames = batchInputFileService.listInputFileNamesWithDoneFile(accountingDocumentBatchInputFileType);
+        LOG.info("createAccountingDocumentsFromXml: Found " + inputFileNames.size() + " files to process");
+        
+        inputFileNames.stream()
+                .forEach(this::processAccountingDocumentFromXml);
+        
+        LOG.info("createAccountingDocumentsFromXml: Finished processing all pending accounting document XML files");
+    }
+
+    protected void processAccountingDocumentFromXml(String fileName) {
+        try {
+            LOG.info("processAccountingDocumentFromXml: Started processing accounting document XML file: " + fileName);
+            
+            byte[] fileData = safelyLoadFileBytes(fileName);
+            AccountingXmlDocumentListWrapper accountingXmlDocuments = (AccountingXmlDocumentListWrapper) batchInputFileService.parse(

```

Comment:
I was trying to conform to the BatchInputFileService interface, since it has a "listInputFileNamesWithDoneFile" method for retrieving the desired file names. (The new JAXB batch input implementation for this PR also uses CUMarshalService behind-the-scenes.) If there's another service or utility for identifying input files that also have a done file, then I could refactor this code accordingly to avoid the extra processing.